javascriptv3/example_code/cross-services/transcribe-streaming-app/src/index.js

```diff
@@ -0,0 +1,208 @@
+- Transcribe speech in real-time using Amazon Transcribe
+- Detect the language of the transcription using Amazon Comprehend
+- Translate the transcription using Amazon Translate
+- Send the transcription and translation by email using Amazon Simple Email Service (Amazon SES)
+*/
+
+// snippet-start:[transcribe.JavaScript.streaming.indexv3]
+import { transcribeClient } from "./libs/transcribeClient.js";
+import { translateClient } from "./libs/translateClient.js";
+import { sesClient } from "./libs/sesClient.js";
+import { comprehendClient } from "./libs/comprehendClient.js";
+import { DetectDominantLanguageCommand } from "@aws-sdk/client-comprehend";
+import { StartStreamTranscriptionCommand } from "@aws-sdk/client-transcribe-streaming";
+import { TranslateTextCommand } from "@aws-sdk/client-translate";
+import { SendEmailCommand } from "@aws-sdk/client-ses";
+import MicrophoneStream from "microphone-stream";
+import getUserMedia from "get-user-media-promise";
+
+// Helper function to encode PCM audio.
+const pcmEncodeChunk = (chunk) => {
+  const input = MicrophoneStream.toRaw(chunk);
+  var offset = 0;
+  var buffer = new ArrayBuffer(input.length * 2);
+  var view = new DataView(buffer);
+  for (var i = 0; i < input.length; i++, offset += 2) {
+    var s = Math.max(-1, Math.min(1, input[i]));
+    view.setInt16(offset, s < 0 ? s * 0x8000 : s * 0x7fff, true);
+  }
+  return Buffer.from(buffer);
+};
+
+window.startRecord = async () => {
+  try {
+    console.log("Recording started");
+    var record = document.getElementById("record");
+    var stop = document.getElementById("stopRecord");
+    record.disabled = true;
+    record.style.backgroundColor = "blue";
+    stop.disabled = false;
+
+    // Start the browser microphone.
+    const micStream = new MicrophoneStream();
+    micStream.setStream(
+      await window.navigator.mediaDevices.getUserMedia({
+        video: false,
+        audio: true,
+      })
+    );
+
+    // Acquire the microphone audio stream.
+    const audioStream = async function* () {
+      for await (const chunk of micStream) {
+        yield {
+          AudioEvent: {
+            AudioChunk: pcmEncodeChunk(
+              chunk
+            ) /* pcm Encoding is optional depending on the source. */,
+          },
+        };
+      }
+    };
+
+    const command = new StartStreamTranscriptionCommand({
+      // The language code for the input audio. Valid values are en-GB, en-US, es-US, fr-CA, and fr-FR.
+      LanguageCode: "en-US",
+      // The encoding used for the input audio. The only valid value is pcm.
+      MediaEncoding: "pcm",
+      // The sample rate of the input audio in Hertz.
+      MediaSampleRateHertz: 44100,
+      AudioStream: audioStream(),
+    });
+
+    // Send the speech stream to Amazon Transcribe.
+    const data = await transcribeClient.send(command);
+    console.log("Success", data.TranscriptResultStream);
+    //return data; //For unit tests only.
```

Comment:
Return (with a space between "//" and Return).

Reply:
done